helm/cluster-api-provider-azure/files/infrastructure/patches/versions/v1beta1/azureclusteridentities.infrastructure.cluster.x-k8s.io.yaml

```diff
@@ -70,17 +70,17 @@
                   type: string
               type: object
             resourceID:
-              description: User assigned MSI resource id.
+              description: ResourceID is the Azure resource ID for the User Assigned MSI resource. Not currently supported.
               type: string
             tenantID:
-              description: Service principal primary tenant id.
+              description: TenantID is the service principal primary tenant id.
               type: string
             type:
-              description: UserAssignedMSI or Service Principal
+              description: Type is the type of Azure Identity used. ServicePrincipal, ServicePrincipalCertificate, or ManualServicePrincipal.
               enum:
                 - ServicePrincipal
                 - ManualServicePrincipal
-                - UserAssignedMSI
+                - ServicePrincipalCertificate
```

Comment:
This looks like a breaking change in theory, but in practice it is not, as `UserAssignedMSI` never worked, see [this comment for more details](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2258/files#r859891486).

In any case we should not be affected, because all vintage workload clusters are using `ServicePrincipal` type.

Reply:
just for reference i did test it in capz 1.6.x and it worked for me using the UA assigned to the MC node

apparently the actual issue with UA Identity was fixed in CAPZ 1.4 https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1104 

:+1: